### PR TITLE
build: update to latest submodule

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
* removed redundant gradle files for documentation and publishing (used submodules)
* update gradle plugin
* add manifest file for test

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by MdUsmanGani — APPROVED** [review]
LGTM

File: remote-config/build.gradle [hunk 1]
```diff
@@ -17,6 +18,13 @@ android {
   defaultConfig {
     consumerProguardFiles 'proguard-rules.pro'
   }
+
+  buildTypes {
+    release {
+      useProguard true
```
Comment (corycaywood) [inline remote-config/build.gradle]: Why is this needed? In my understanding, this setting is only used during the build of the AAR and won't affect the end user. Is that wrong?

The `consumerProguardFiles` option is used for setting the proguard file which is bundled into the AAR.

File: remote-config/build.gradle [hunk 2]
```diff
@@ -1,4 +1,5 @@
 apply from: '../config/android/library.gradle'
+apply from: '../config/documentation/dokka/android.gradle'
```
Comment (corycaywood) [inline remote-config/build.gradle]: I think this requires some additional setup after updating to the new version, right? There needs to be a `dokka` block which configures the source directory: https://github.com/rakutentech/android-buildconfig/tree/master/documentation#dokka

File: remote-config/build.gradle (hunk 1 above)
Comment (vincentdelmundo) [reply remote-config/build.gradle]: Removed. This is unnecessary.

File: remote-config/build.gradle (hunk 2 above)
Comment (vincentdelmundo) [reply remote-config/build.gradle]: Added setting of source directory

**Review by corycaywood — APPROVED** [review]
Looks good 👍
